src/Audio.md

```diff
@@ -0,0 +1,110 @@
+
+The APU processes a lot of events at a fixed rate dependent on the event's type.
+All interfaces to the APU use **durations** instead of frequencies, which may be confusing as signal theory and music are more typically based on the latter.
+Thus, durations will be expressed from their frequencies: for example, a "256 Hz tick" means "1 ∕ 256th of a second".
+
+The length of APU ticks is not affected by [CGB double speed](<#FF4D — KEY1 (CGB Mode only): Prepare speed switch>), so the APU works just the same regardless of CPU speed.
+
+::: warning
+
+The Game Boy's APU is actually full of tricky details; this chapter will mostly describe the intended / common behavior, and often paper over bugs & quirks.
+Readers wishing to learn more should read the [APU details](<#Audio Details>) chapter.
+
+:::
+
+### Triggering
+
+**Triggering** a channel causes it to turn on if it wasn't[^trig_dac_off], and to start playing its wave from the beginning[^pulse_restart].
+Most changes to a channel's parameters take effect immediately, but some require re-triggering the channel.
+
+### Volume & envelope
+
+The volume can be controlled in two ways: there is a "master volume" control[^vol_knob] (which has separate settings for the left and right outputs), and each channel's volume can be individually set as well (CH3's less precisely than the others).
+
+Additionally, an [**envelope**](https://en.wikipedia.org/wiki/Envelope_(music)) can be configured for all channels except CH3, which allows automatically adjusting the volume over time.
+The parameters that can be controlled are how many times the envelope will be stepped, in which direction, and how long between steps.
+All envelopes are ticked at 64 Hz, and every 1–7 of those ticks, the volume will be increased or decreased.
+
+### Length timer
+
+All channels can be individually set to automatically shut themselves down after a certain amount of time.
+
+If the functionality is enabled, a channel's **length timer** ticks up[^len_cnt_dir] at 256 Hz (tied to [DIV-APU](<#DIV-APU>)) from the value it's initially set at.
+When the length timer reaches 64, the channel is turned off.
+
+### Frequency
+
+Music notes and audio waves are typically manipulated in terms of **frequency**[^pitch], i.e. how often the signal repeats per second.
+However, as explained above, the Game Boy APU primarily works with durations; thus, **wavelengths** will be used instead of frequency.[^len_raw]
+
+::: warning
+
+The term "wavelength" throughout this document does *not* designate the inverse of the frequency, but instead a quantity akin to it.
+See the description of each NRx3 register for more information.
+
+:::
+
+---
+
+[^trig_dac_off]:
+If [the channel's DAC](#DACs) is off, the channel will not turn on.
+
+[^pulse_restart]:
+Except for pulse channels, whose index into their waves is only ever reset by turning the APU off.
+This is not noticeable when playing sound, and only matters for "advanced" APU usage.
```

Comment:
Following the theme of not describing things solely in terms of primitives, "index into their waves" should probably be "phase" or "phase position". Also, I'm not sure this information should be hidden away in a footnote since it applies to half of the available channels. Also, not sure exactly what counts as advanced APU usage, but a fairly common trick is to play two slightly detuned notes on two channels, which can sound different every time due to this effect.

Reply:
This is my turn being confused—what does "detuned" mean?

I think "phase position" would best describe the intent here.

Reply:
Detuned just means the two notes have a slightly different frequency. The effect in question happens when two such notes create a "phasing" sound with each other. A simple way to illustrate the concept is for example using my bgbtest.gb ROM (which comes with BGB.) Press one of the buttons quickly repeatedly. Every second note is detuned, and if you do it quick enough, they overlap and create this effect.

What you can't do in a very predictable way, is an effect similar to a PWM that goes from a low percent duty to a 50% duty, since the phase position might be "random" on every note retrigger. I can make a sound example later.

Reply:
Agree here, I'd avoid saying "index into their waves"

Reply:
How is it now?

Reply:
Ok to me, but I opened https://github.com/gbdev/pandocs/issues/445 to have the chance to continue this conversation